lib/rrule.js

```diff
@@ -2209,13 +2209,39 @@
       }
     },
 
+    /**
+     * Filters out floating timezone properties.
+     * @param {String} s A string containing the set of recurrence properties.
+     * @return {String}  A filtered string containing the set of recurrence properties without floating timezones.
+     */
+    filterFloatingTimezones: function (s) {
+      // Uses a regex to split the recurrence string by carriage return.
+      var parsedRecurrenceString = s.split(/\s+/)
+
+      // Remove any of the recurrence date properties that contain floating timezone properties.
+      parsedRecurrenceString.forEach(function (ruleString) {
+        var index = parsedRecurrenceString.indexOf(ruleString);
+
+        if (contains(ruleString, 'TZID')) {
+          if (index !== -1) {
+            parsedRecurrenceString.splice(index, 1)
+          }
+        }
+      })
+
+      return parsedRecurrenceString.join('\n')
+    },
+
     parse: function (s, options) {
       options = options || {}
 
       var invalid = []
       var keys = Object.keys(options)
       var defaultKeys = Object.keys(RRuleStr.DEFAULT_OPTIONS)
 
+      // Sets the recurrence string to the original recurrence model without any floating timezones.
+      s = this.filterFloatingTimezones(s)
```

Comment:
Rather than mutating the original param, why don't we keep it handy? Think:

```javascript
parse: function (rawString, options) {
// ...
// Terrible name, but you get idea.
var safeString = this.filterFloatingTimezones(rawString);
```